src/internal/v1api/modules.go

```diff
@@ -0,0 +1,128 @@
+package v1api
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"log/slog"
+	"os"
+	"path/filepath"
+
+	"registry-stable/internal"
+	"registry-stable/internal/module"
+)
+
+// GenerateModuleResponses generates the response for the module version listing API endpoints.
+// For more information see
+// https://opentofu.org/docs/internals/module-registry-protocol/#list-available-versions-for-a-specific-module
+// https://opentofu.org/docs/internals/module-registry-protocol/#download-source-code-for-a-specific-module-version
+func (g Generator) GenerateModuleResponses(_ context.Context, namespace string, name string, targetSystem string) error {
+	logger := slog.With(slog.String("namespace", namespace), slog.String("name", name), slog.String("targetSystem", targetSystem))
+
+	// TODO: Get path calculation from somewhere else
+	path := filepath.Join(namespace[0:1], namespace, name, targetSystem+".json")
+
+	metadata, err := g.readModuleMetadata(path, logger)
+	if err != nil {
+		return err
+	}
+
+	// Right now the format is pretty much identical, however if we want to extend the results in the future to include
+	// more information, we can do that here. (i.e. the root module, or the submodules)
+	versionsResponse := make([]VersionResponseItem, len(metadata.Versions))
+	for i, m := range metadata.Versions {
+		versionsResponse[i] = VersionResponseItem{Version: m.Version}
+
+		err := g.writeModuleVersionDownload(namespace, name, targetSystem, m.Version)
+		if err != nil {
+			return fmt.Errorf("failed to write metadata version download file for version %s: %w", m.Version, err)
+		}
+		logger.Debug("Wrote metadata version download file", slog.String("version", m.Version))
+	}
+
+	// Write the /versions response
+	err = g.writeModuleVersionListing(namespace, name, targetSystem, versionsResponse)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// readModuleMetadata reads the module metadata file from the filesystem directly. This data should be the data fetched from the git repository.
+func (g Generator) readModuleMetadata(path string, logger *slog.Logger) (*module.MetadataFile, error) {
+	// list directories at the root of the fs
+	dirs, err := fs.ReadDir(g.ModuleFS, ".")
+	if err != nil {
+		slog.Error("Failed to list directories", slog.Any("err", err))
+		os.Exit(1)
+	}
+
+	for _, d := range dirs {
+		slog.Info("Found directory", slog.String("dir", d.Name()))
+	}
```

Comment:
I'm guessing this is only temporary for debug purposes?

Reply:
Oops, yes ill remove this.